packages/amplify_datastore_plugin_interface/lib/src/types/query/query_sort.dart

```diff
@@ -30,7 +30,7 @@ class QuerySortBy {
   final QuerySortOrder order;
   final String field;
 
-  const QuerySortBy({this.order, this.field});
+  const QuerySortBy({required this.order, required this.field});

```

Comment:
Why are we going from optiona -> required?

Reply:
Same as the above reason, all usages of this constructor method pass the fields: 

```
  // Sorting (Discussed later in this doc)
  QuerySortBy ascending() {
    return new QuerySortBy(field: fieldName, order: QuerySortOrder.ascending);
  }

  QuerySortBy descending() {
    return new QuerySortBy(field: fieldName, order: QuerySortOrder.descending);
  }
```

Reply:
This won't break any customers?

Reply:
It will cause compile errors for customers not passing those fields.  However, aren't these fields require in the native side as well.  For example here are the constructors for Android: 

```
    public QuerySortBy(@NonNull String field, @NonNull QuerySortOrder sortOrder) {
        this(null, field, sortOrder);
    }

    /**
     * Constructor for {@code QuerySortBy}.
     *
     * @param modelName name of the model being sorted.
     * @param field name of field to sort by.
     * @param sortOrder order to sort by, either ASCENDING or DESCENDING.
     */
    public QuerySortBy(@Nullable String modelName, @NonNull String field, @NonNull QuerySortOrder sortOrder) {
        this.modelName = modelName;
        this.field = Objects.requireNonNull(field);
        this.sortOrder = Objects.requireNonNull(sortOrder);
    }
```

Reply:
Hmm ok, as long as this is the case for both native libraries (and we mark this as breaking change - we should specifically call out all breaking changes in the change log and possibly in a pinned issue).